dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,281 @@
+        content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+        using var request = new HttpRequestMessage(HttpMethod.Post, url);
+        this.AddDefaultHeaders(request);
+        request.Content = content;
+        using var response = await this.SendWithSuccessCheckAsync(this.HttpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAndTranslateExceptionAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        return json.RootElement.GetProperty("upload_url").GetString()
+               ?? throw new KernelException("Property 'upload_url' expected but not found.");
+    }
+
+    private async Task<string> CreateTranscriptAsync(
+        string audioUrl,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url("v2/transcript");
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
+        {
+            foreach (var attribute in executionSettings.ExtensionData)
+            {
+                jsonRequest[attribute.Key] = JsonValue.Create(attribute.Value);
+            }
+        }
+
+        using var request = HttpRequest.CreatePostRequest(url, jsonRequest);
+        this.AddDefaultHeaders(request);
+        using var response = await this.SendWithSuccessCheckAsync(this.HttpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new KernelException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(
+        string transcriptId,
+        PromptExecutionSettings? executionSettings,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url($"v2/transcript/{transcriptId}");
+
+        var pollingInterval = TimeSpan.FromMilliseconds(500);
+        if (executionSettings is AssemblyAIAudioToTextExecutionSettings aaiSettings)
+        {
+            pollingInterval = aaiSettings.PollingInterval;
+        }
+
+        while (true)
+        {
+            ct.ThrowIfCancellationRequested();
+            using var request = HttpRequest.CreateGetRequest(url);
+            this.AddDefaultHeaders(request);
+            using var response = await this.SendWithSuccessCheckAsync(this.HttpClient, request, ct).ConfigureAwait(false);
+            var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
```

Comment:
```suggestion
            using var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
```
I know disposing response dispose also stream, but good practice is always call dispose if something is disposable.